File: mobile/library/swift/src/RequestTrailers.swift
import Foundation

/// Trailers representing an outbound request.
@objcMembers
public final class RequestTrailers: Trailers {
  /// Convert the trailers back to a builder for mutation.
  ///
  /// - returns: The new builder.
  public func toRequestTrailersBuilder() -> RequestTrailersBuilder {
    return RequestTrailersBuilder(headers: self.headers)
  }
}

File: mobile/library/kotlin/io/envoyproxy/envoymobile/RequestTrailers.kt
package io.envoyproxy.envoymobile

/** Trailers representing an outbound request. */
@Suppress("EmptyClassBlock")
class RequestTrailers : Trailers {
  /**
   * Internal constructor used by builders.
   *
   * @param trailers: Trailers to set.
   */
  internal constructor(trailers: Map<String, List<String>>) : super(trailers)

  /**
   * Instantiate a new builder.
   *
   * @param container: The headers container to start with.
   */
  internal constructor(container: HeadersContainer) : super(container)

  /**
   * Convert the trailers back to a builder for mutation.
   *
   * @return RequestTrailersBuilder, The new builder.
   */
  fun toRequestTrailersBuilder() = RequestTrailersBuilder(container)
}
